libnmstate/nm/wired.py

```diff
@@ -92,24 +93,24 @@ def create_setting(iface_state, base_con_profile):
             nm_wired_setting.props.duplex = None
 
         elif not setting.speed:
-            ethtool_results = minimal_ethtool(str(iface_state['name']))
-            setting.speed = ethtool_results['speed']
+            ethtool_results = minimal_ethtool(str(iface_state[Interface.NAME]))
+            setting.speed = ethtool_results[Ethernet.SPEED]
         elif not setting.duplex:
-            ethtool_results = minimal_ethtool(str(iface_state['name']))
-            setting.duplex = ethtool_results['duplex']
+            ethtool_results = minimal_ethtool(str(iface_state[Interface.NAME]))
+            setting.duplex = ethtool_results[Ethernet.DUPLEX]
 
     elif setting.auto_negotiation is False:
         nm_wired_setting.props.auto_negotiate = False
-        ethtool_results = minimal_ethtool(str(iface_state['name']))
+        ethtool_results = minimal_ethtool(str(iface_state[Interface.NAME]))
         if not setting.speed:
-            setting.speed = ethtool_results['speed']
+            setting.speed = ethtool_results[Ethernet.SPEED]
         if not setting.duplex:
-            setting.duplex = ethtool_results['duplex']
+            setting.duplex = ethtool_results[Ethernet.DUPLEX]
 
     if setting.speed:
         nm_wired_setting.props.speed = setting.speed
 
-    if setting.duplex in ['half', 'full']:
+    if setting.duplex in [Ethernet.DUPLEX_HALF, Ethernet.DUPLEX_FULL]:
```

Comment:
nit: IMHO `HALF_DUPLEX` is more natural. If you like to have a hierachy, `DUPLEX.HALF` would be better IMHO. (not a must)

Reply:
Your proposal feels both right and wrong, and I cannot say which wins. But it is worth changing if it is worth commenting about :)
Done